### PR TITLE
Convert chef-server-ctl key commands to use Chef::Key

### PR DESCRIPTION
**Pull request by tylercloke**
:warning: *THIS WILL FLOAT CHEF-CLIENT ON MASTER* :warning: 
+ Update `chef-server-ctl` key commands to use `Chef::Key` and support server-side key generation.
+ Float knife-opc on master.
+ Float chef-client on master (makes sense to do now, since I need functionality from master I just added).

Depends on:
+ https://github.com/chef/omnibus-software/pull/403
+ https://github.com/chef/omnibus-chef/pull/372

Manually test:
- [x] `knife-opc` related commands
- [x] `chef-client` related ctl commands

TODO:
- [x] Test is https://github.com/chef/knife-opc/issues/31 still exists in build with `knife-opc` 0.3.1. Close if not.
- [x] Resolve "looks like we need a new_record/5 function in the chef_key code for it to work with the new api version stuff" -SSD
- [x] Changelog

[Passing Build](http://wilson.ci.chef.co/job/chef-server-12-trigger-ad_hoc/480/downstreambuildview/)

Ping @chef/lob

### CODE REVIEW

File: files/private-chef-ctl-commands/helpers/key_ctl_helper.rb [hunk 1]
```diff
@@ -31,25 +32,27 @@ def exit_failure(msg)
...
+  def populate_client_key(clientname, name, public_key, expiration_date)
+    key = Chef::Key.new(clientname, "client")
+    populate_key_helper(key, name, public_key, expiration_date)
   end
 
-  def pivotal_config
-    "/etc/opscode/pivotal.rb"
+  def populate_user_key(username, name, public_key, expiration_date)
+    key = Chef::Key.new(username, "user")
+    puts "wtf"*100
```
Comment (stevendanna) [inline files/private-chef-ctl-commands/helpers/key_ctl_helper.rb]: Guessing this is leftover from debugging :)

Comment (tylercloke) [inline files/private-chef-ctl-commands/helpers/key_ctl_helper.rb]: Naw, completely intentional :P Good catch.

File: files/private-chef-ctl-commands/helpers/key_ctl_helper.rb [hunk 2]
```diff
@@ -71,21 +76,13 @@ def read_and_check_key(key_path)
...
-    end
-  end
-
   def add_key_usage
-    "\nUsage: Expiration date defaults to infinity. Pass an ISO 8601 fomatted string: YYYY-MM-DDTHH:MM:SSZ e.g. 2013-12-24T21:00:00Z in UTC timezone\nUsage: Default name used is the fingerprint of the key passed."
+<<EOS
+
+Usage: If --public-key-path isn't passed, the server will generate a public key for you.
+Usage: Expiration date defaults to infinity. Pass an ISO 8601 fomatted string: YYYY-MM-DDTHH:MM:SSZ e.g. 2013-12-24T21:00:00Z in UTC timezone.
+Usage: Default name used is the fingerprint of the key passed.
```
Comment (marcparadise) [inline files/private-chef-ctl-commands/helpers/key_ctl_helper.rb]: What's default name in case no key is passed?

**Comment by marcparadise** [comment]
This looks good to me other than the question above, and a reminder to remove dev branches.

File: files/private-chef-ctl-commands/helpers/key_ctl_helper.rb (hunk 2 above)
Comment (tylercloke) [inline files/private-chef-ctl-commands/helpers/key_ctl_helper.rb]: There isn't one. You need to either pass `--public-key-path` or `--key-name`.

Comment (marcparadise) [inline files/private-chef-ctl-commands/helpers/key_ctl_helper.rb]: :+1:

**Pull request merged**